src/modules/commander/Commander.hpp

```diff
@@ -155,27 +171,26 @@ class Commander : public ModuleBase<Commander>, public ModuleParams
 
 	void mission_init();
 
-	/**
-	 * Update the telemetry status and the corresponding status variables.
-	 * Perform system checks when new telemetry link connected.
-	 */
-	void poll_telemetry_status();
-
 	/**
 	 * Checks the status of all available data links and handles switching between different system telemetry states.
 	 */
-	void data_link_checks(int32_t highlatencydatalink_loss_timeout, int32_t highlatencydatalink_regain_timeout,
-			      int32_t datalink_loss_timeout, int32_t datalink_regain_timeout, bool *status_changed);
-
-	// telemetry variables
-	struct telemetry_data {
-		int subscriber = -1;
-		uint64_t last_heartbeat = 0u;
-		uint64_t last_dl_loss = 0u;
-		bool preflight_checks_reported = false;
-		bool lost = true;
-		bool high_latency = false;
-	} _telemetry[ORB_MULTI_MAX_INSTANCES];
+	void		data_link_check(bool &status_changed);
+	int		_telemetry_status_sub[ORB_MULTI_MAX_INSTANCES] {};
+
+	uint64_t	_datalink_last_heartbeat_gcs{0};
```

Comment:
These should all be `hrt_abstime`

Reply:
Done here 1bb3b3fcfdc8bf375186b4a78c8e3052fceb933a